### DISCUSSION TITLE
Logrus log wrapper doesn't honor levels

### DISCUSSION THREAD

**Discussion by jacksontj** [go-kit/kit#1151] [Q&A]
Recently changes were made to the logrus log wrapper such that you can define a log level that all messages will be written at, but in reality logs have a level defined on the message (not at the logger). So the current logger is insufficient. I have created a simple wrapper (https://github.com/jacksontj/promxy/pull/310/commits/8069dfb38847d2223415925ae8f94425e6aa5a04) which does honor the log levels, would you be open to a PR with this inclusion? If so, would this be a new wrapper or an added option to the current one (if so, how would we configure that -- likely another option that sets some bool saying "honor log level")

**Comment by ChrisHines** [comment]
My apologies for the slow response.

I agree it would be nice to integrate the `log/levels` package to the `log/logrus` adapter more comprehensively.

In general I consider the Go kit log/logrus package a best effort community contribution. I haven't put any effort into making a high fidelity adapter simply because the use case seems dubious to me. But I'll happily take improvements. Submit a PR and I'll review it.

There is some previous discussion about this and another related issue at #929 if you haven't seen that. Maybe you can take a look at that and see if you have any ideas how to solve both issues.
